examples/auction/src/lib.rs

```diff
@@ -95,19 +83,29 @@ enum FinalizeError {
 
 /// Init function that creates a new auction
 #[init(contract = "auction", parameter = "InitParameter")]
-fn auction_init(ctx: &impl HasInitContext) -> InitResult<((), State)> {
+fn auction_init<S: HasStateApi>(
+    ctx: &impl HasInitContext,
+    state_builder: &mut StateBuilder<S>,
+) -> InitResult<State<S>> {
     let parameter: InitParameter = ctx.parameter_cursor().get()?;
-    Ok(((), fresh_state(parameter.item, parameter.expiry)))
+    let state = State {
+        auction_state: AuctionState::NotSoldYet,
+        highest_bid:   Amount::zero(),
+        item:          parameter.item,
```

Comment:
I think we should make the `item` a `String` now that we're at it.

Reply:
Changed.